tests/baselines/reference/jsFileAlternativeUseOfOverloadTag.js

```diff
@@ -175,4 +175,17 @@ declare namespace example3 {
  * function (error, result)
  *     If callback is provided it will be called with evaluation result
  */
-type callback = (error: any, result: any) => any;

```

Comment:
This change isn't great but tbh the behavior for a scenario like this looks underspecified to me. We can even read this at the top of the relevant test file:
```ts
// These are a few examples of existing alternative uses of @overload tag.
// They will not work as expected with our implementation, but we are
// trying to make sure that our changes do not result in any crashes here.
```

And even here we can't type this implicit~ `callback` type in full as it complains about "'return' tag already specified.(1223)"
```ts
const example4 = {
  /**
   *   @param options {Record<string, any>}
   *   @param callback {callback}
   *   @return {string}
   *   @callback callback
   *     @param error {Error}
   *     @param result {string}
   *     @return {number}
   */
  evaluate: function evaluate(options, callback) {
    return ''
  },
};
```